app/javascript/components/Index.js

```diff
@@ -0,0 +1,27 @@
+import React from "react"
+import PropTypes from "prop-types"
+
+import App from "./App"
+import { BrowserRouter as Router } from "react-router-dom"
+import { ContextProvider } from "./context/Context"
+
+function Index(props) {
+
+    const {
+      logged_in,
+      sign_in_route,
+      sign_out_route,
+      current_user_id
+    } = props
+
+    return (
+        <ContextProvider>  
+            <Router>
```

Comment:
We don't want to add a new Router here.  Only one router per app is required, and everything rendered inside of it uses the main one up in the state tree.